### DISCUSSION TITLE
Updating gpt model in config not working

### DISCUSSION THREAD

**Discussion by ben-wall** [jackMort/ChatGPT.nvim#343] [Q&A] [answered]
I update my model in my config like so 
```
        openai_params = {
          model = "gpt-4",
          frequency_penalty = 0,
          presence_penalty = 0,
          max_tokens = 800,
          temperature = 0,
          top_p = 1,
          n = 1,
        },
        openai_edit_params = {
          model = "gpt-4",
          temperature = 0,
          top_p = 1,
          n = 1,
        },
```
...which I have verified as being read properly by changing another option and seeing that update.

when I press ctrl o to open the options it says gpt-4 (which I by the way cannot get to change back to gpt-3.5-turbo, now) but when I ask 'what model of chat gpt am I using' it always reports chat gpt 3.

I've tried even trying to break the model by changing it to 'gpt-5' and it still seems to be using gpt 3 and doesn't break.

**Comment by amplicity** [comment]
I have the same issue

**Comment by ysqander** [comment]
I have the same issue. I think the default actions specify 3.5 Turbo and override the params in the config. 

When I wrote a simple custom function with the model gpt-4-1106-preview, I received a reply from gpt-4 with cutoff early 2023.

**Comment by david-strejc** [comment]
Same here - is there a way how to change model to gpt-4-1106-preview?

**Comment by winslowb** [comment]
i've been sitting with the same issue for months and months

**Comment by ben-wall** [comment]
> I have the same issue. I think the default actions specify 3.5 Turbo and override the params in the config.
> 
> When I wrote a simple custom function with the model gpt-4-1106-preview, I received a reply from gpt-4 with cutoff early 2023.

how did you even get it to gpt-4? that's all I want

**Comment by ysqander** [comment]
@ben-wall 

I changed the config to this 

config = function()
      require("chatgpt").setup({
        api_key_cmd = "dcli note opk",
        openai_params = {
          model = "gpt-4",
          frequency_penalty = 0,
          presence_penalty = 0,
          max_tokens = 600,
          temperature = 0,
          top_p = 1,
          n = 1,
        },
        openai_edit_params = {
          model = "gpt-4",
          frequency_penalty = 0,
          presence_penalty = 0,
          temperature = 0,
          top_p = 1,
          n = 1,
        },
        actions_paths = { vim.fn.expand("~/GPTnvimActions/actions.json") },

Then in my custom actions.json I have the same default action but changing the model from 3.5 to gpt-4-1106-preview. For example:

  "complete_code_gpt4Turbo": {
    "type": "chat",
    "opts": {
      "template": "Complete the following code written in {{lang}} by pasting the existing code and continuing it.\n\nExisting code:\n```{{filetype}}\n{{input}}\n```\n\n```{{filetype}}\n",
      "strategy": "replace",
      "params": {
        "model": "gpt-4-1106-preview",
        "stop": [
          "```"
        ]
      }
    }

I tested it with an action that asked it which model it is and the cutoff and it said GPT4 and cutoff early 2023. So this works only for actions. The main chat window is still hooked to 3.5 despite the config changes.

**Comment by e2r2fx** [comment, marked as answer]
this is because the plugin tries to read from a file in the home directory, depending on the type of settings. Look for .chatgpt-chat_completions-params.json and remove it from your home. Then the configuration should kick in.

**Reply by gipo355** [reply]
wow, isn't this worth adding to the README? I didn't know it was creating/reading files in ~ and was stuck on this for months.

I think it deserves a note @jackMort

**Comment by JeancarloBarrios** [comment]
I have the same issue, and I do not have the file  .chatgpt-chat_completions-params.json . Is there any other suggestion?

**Comment by winslowb** [comment]
[](url)for me...i have a ~/.chatgpt-chatcompletions-params.json ; 
{"max_tokens":3000,"n":1,"top_p":1,"temperature":0,"presence_penalty":0,"frequency_penalty":0,"model":"gpt-4-turbo"}

![Screenshot from 2024-05-02 09-24-38](https://github.com/jackMort/ChatGPT.nvim/assets/71040996/23dba734-d9df-4f99-a443-ca6e79067213)

**Comment by h4ckm1n-dev** [comment]
i have the same problem using ollama as an endpoint... some windows still use gtp3.5 turbo

EDIT :  It's work ! 
for Lazyvim user the json file is located in "~/.local/share/nvim/lazy/ChatGPT.nvim/lua/chatgpt/flows/actions"
